### PR TITLE
added error handling functions

### PR DESCRIPTION
**Pull request by cjc8**
Hopefully this is closer to what you envisioned!

### CODE REVIEW

**Review by schrockblock — COMMENTED** [review]
I like this a lot. However, there's a lot of repetition (which isn't unreasonable, just confusing initially) – could you add a playground to show how it works in its various iterations?

**Review by schrockblock — APPROVED** [review]
I think this is working pretty nicely in Tapper, so let's get some of the added helper functions from there moved into this

File: FunNet/Classes/Core/FunFuncErrorHandling.swift [hunk 1]
```diff
@@ -195,3 +196,50 @@ public func responseString(messageMap: [Int:String]) -> (HTTPURLResponse) -> Str
...
+            if let vc = presenter, let dataString = error.userInfo["data"] as? String, let data = dataString.data(using: .utf8), let responseError = JsonProvider.decode(T.self, from: data), let message = unwrapper(responseError) {
+                codeStringAlert(code: error.code, description: message) ?> vc.presentClosure()
+            } else {
+                error |> debugServerErrorHandler(vc: presenter, errorMap: urlResponseErrorMessages << overrideCodes)
+            }
+        }
+    }
+}
+
+infix operator <<: AdditionPrecedence
```
Comment (schrockblock) [inline FunNet/Classes/Core/FunFuncErrorHandling.swift:228]: Should we put this in LithoOperators? I'm honestly asking, since I'm not sure myself.

**Review by schrockblock — APPROVED** [review]
One question, but otherwise I think this is pretty great

File: FunNet/Classes/Core/FunFuncErrorHandling.swift (hunk 1 above)
Comment (cjc8) [reply FunNet/Classes/Core/FunFuncErrorHandling.swift:228]: we should, i like it. i will do that

Comment (cjc8) [reply FunNet/Classes/Core/FunFuncErrorHandling.swift:228]: ah it's already in there it sees like. nvm

**Pull request merged**